src/core/lib/OrderValidator.sol

```diff
@@ -412,7 +412,7 @@ contract OrderValidator is Executor, ZoneInteraction {
         // Retrieve the order status using the derived order hash.
         OrderStatus storage orderStatus = _orderStatus[orderHash];
 
-        bool hasCarryOrNoNumerator = numerator == 0;
```

Comment:
I think it's unnecessary because `updateStatus` will never be called with a numerator of 0. At least when called from `match/fulfillAvailable -> _validateOrdersAndPrepareToFulfill` if the validation phase returns a numerator of 0, we don't set `orderHashes[i]`, and then this item is also skipped in the update status loop.

However, it looks possible that `_updateStatus` is called with a numerator of 0 when called from `fulfill(Advanced)Order -> _validateAndFulfillAdvancedOrder -> _updateStatus`. Need to review this.

Reply:
we catch numerator == 0 early in [`_validateOrder`](https://github.com/ProjectOpenSea/seaport-1.6/blob/f0f38582701f0a6bcce207c4ccb6e0e9fea09e56/src/core/lib/OrderValidator.sol#L206). It can only return `0` if `revertOnInvalid=false` and `_verifyTime` or `_verifyOrderStatus` fails. So should be fine for all paths.